pkg/api/convertertoapi.go

```diff
@@ -307,7 +382,8 @@ func convertV20160930VMDiagnostics(v20160930 *v20160930.VMDiagnostics, api *VMDi
 }
 
 func convertV20160330DiagnosticsProfile(v20160330 *v20160330.DiagnosticsProfile, api *DiagnosticsProfile) {
-	convertV20160330VMDiagnostics(&v20160330.VMDiagnostics, &api.VMDiagnostics)
+	api.VMDiagnostics = &VMDiagnostics{}
+	convertV20160330VMDiagnostics(v20160330.VMDiagnostics, api.VMDiagnostics)
```

Comment:
stupid GitHub doesn't allow me to leave comments elsewhere. I meant nil check on v20160330.VMDiagnostics inside convertV20160330VMDiagnostics